File: src/main/kotlin/chapter/chaptet7/Chapter7_2_compareTo.java
package chapter.chaptet7;

public class Chapter7_2_compareTo {

    static class JavaPerson implements Comparable {
        public int age;

        public JavaPerson(int age) {
            this.age = age;
        }

        @Override
        public int compareTo(Object o) {
            return age >= ((JavaPerson)o).age ? 1 : 0;
        }
    }

    public static void main(String[] args) {
        JavaPerson person1 = new JavaPerson(10);
        JavaPerson person2 = new JavaPerson(20);

        System.out.println(person1.compareTo(person2));

        // 이렇게는 안됨
        // System.out.println(person1 > person2);
    }


}

File: src/main/kotlin/chapter/chapter7/Chapter7_2_compareTo.kt
package chaptet7

class Person3(
    val age: Int
) : Comparable<Person3> {
    override fun compareTo(other: Person3): Int {
        return compareValuesBy(this, other, Person3::age)
    }
}

fun main() {
    val person1 = Person3(10)
    val person2 = Person3(20)
    println(person1 < person2)
}
